src/mount/bsd.rs

```diff
@@ -396,14 +396,7 @@ impl<'a> Nmount<'a> {
         match Errno::result(res) {
             Ok(_) => Ok(()),
             Err(error) => {
-                let errmsg = match errmsg.iter().position(|&x| x == 0) {
-                    None => None,
-                    Some(0) => None,
-                    Some(n) => {
-                        let sl = &errmsg[0..n + 1];
-                        Some(CStr::from_bytes_with_nul(sl).unwrap())
-                    }
-                };
+                let errmsg = CStr::from_bytes_until_nul(&errmsg[..]).ok();
```

Comment:
The behavior slightly changes compared to the previous code, when `errmsg[0]` is `0`, the new impl will return `Some(an_empty_CStr)` rather than `None`, but it is still consistent with the C code behavior I guess

Reply:
Good catch.  The C code in the mount command matches the old behavior, so I'll restore it.